### ISSUE TITLE
Example for registering txt records

### ISSUE THREAD

**Issue by ruurdadema** [stbuehler/rust-async-dnssd#5]
Awesome library! Thanks for that!

I'm able to register a a service using `async_dnssd::register` which works fine!

However, I don't see a way to add a txt record to the service. Whatever I try ends up in a vague error like: `kind: Other, error: known error Invalid: invalid`.

Could you maybe provide an example on how to register a service with txt record?

**Comment by stbuehler** [comment]
Hm, I'm suprised `register` is working for you (because it seems I mixed some channel types when casting back and forth with the C callback API).

Anyway, I added further "entry points" in the crate doc, and also added a table with a mapping from C API to the crate; those should explain how to register "raw" records (which is what the title of this issues seems to ask for).

The `txt` parameter when registering a service needs to be valid "wire encoded" RDATA for a `TXT` record (or be empty); the [dnssd `DNSServiceRegister`](https://developer.apple.com/documentation/dnssd/1804733-dnsserviceregister) manual explains this for the parameter `txtRecord`.

I might change this to something easier to use (dnssd provides "TXT Record Construction Functions"; I don't intend to wrap them, they should be easy enough to provide in native rust).
